lib/model/folder.go

```diff
@@ -51,3 +62,105 @@ func (f *folder) scanSubdirsIfHealthy(subDirs []string) error {
 	}
 	return nil
 }
+
+// A function to provide the ability to validate and modify local changes,
+// before they are committed to the database
+// Default behavior is to apply the changes as-is, overwrite function as needed
+func (f *folder) validateAndUpdateLocalChanges(fs []protocol.FileInfo) []protocol.FileInfo {
```

Comment:
I'm not super keen on this method. It gets called from `model` (where it used to call `updateLocals`) then does something (potentially) and calls back into the model while returning the  same thing it was passed, maybe. Would this not be cleaner as just a `filterLocalChanges()` with the same signature, that just modified the list of changes as appropriate and then let the model do `updateLocals` as usual? Easier to test too.